dace/codegen/targets/intel_fpga.py

```diff
@@ -404,10 +412,23 @@ def generate_kernel_internal(self, sdfg, state, kernel_name, subgraphs,
 
         self.generate_modules(sdfg, state, kernel_name, subgraphs,
                               subgraph_parameters, sc_parameters,
-                              symbol_parameters, kernel_stream,
+                              symbol_parameters, kernel_body_stream,
                               host_code_header_stream, host_code_body_stream)
 
-        kernel_stream.write("\n")
+        kernel_body_stream.write("\n")
+
+        # Generate data width converters
+        self.generate_converters(sdfg, kernel_header_stream)
+
+        res = (kernel_header_stream.getvalue() + kernel_body_stream.getvalue())
+
+        # TODO: To avoid re-implementing process_out_memlets and potentially
+        # other CPU codegen functionality, convert to OpenCL vector types here.
+        # It was either this or copy-pasting large amounts of code :-)
+        # We need a solution for this in the new code generator.
+        res = re.sub("dace::vec<([^,]+), ([2-9]+|[1-9][0-9]+)>", "\\1\\2", res)
```

Comment:
This won't work with `unsigned long`, for instance (which should be ulong4) or `long long` (longlong2)
the opencl type is planned to be returned in dtypes.

Reply:
Right, so do I implement this here? I can't access the dtype for the reason in the comment

Reply:
in dtypes.py we can have an `opencltype` (or `ctype`)?

Reply:
Yes, I can take a look at implementing something like this, but I still can't implement this in a "correct" way (i.e. without the silly regex) before we have something like a global `make_read` and `make_write`.

Reply:
I've removed this for now, and left it for a future PR as part of new codegen.